src/test/codewhisperer/service/codewhisperer.test.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import sinon from 'sinon'
+import { anyString, spy } from '../../utilities/mockito'
+import { codeWhispererClient } from '../../../codewhisperer/client/codewhisperer'
+import CodeWhispererUserClient, {
+    SendTelemetryEventResponse,
+    TelemetryEvent,
+} from '../../../codewhisperer/client/codewhispereruserclient'
+import globals from '../../../shared/extensionGlobals'
+import { AWSError, Request, Service } from 'aws-sdk'
+import { DefaultAWSClientBuilder, ServiceOptions } from '../../../shared/awsClientBuilder'
+import { FakeAwsContext } from '../../utilities/fakeAwsContext'
+import userApiConfig = require('./../../../codewhisperer/client/user-service-2.json')
+import { AuthUtil } from '../../../codewhisperer/util/authUtil'
+
+describe('codewhisperer', async function () {
+    let clientSpy: CodeWhispererUserClient
+    let telemetryEnabledDefault: boolean
+
+    beforeEach(async function () {
+        sinon.restore()
+        const builder = new DefaultAWSClientBuilder(new FakeAwsContext())
+        clientSpy = spy(
+            (await builder.createAwsService(Service, {
+                apiConfig: userApiConfig,
+            } as ServiceOptions)) as CodeWhispererUserClient
+        )
+        sinon.stub(codeWhispererClient, 'createUserSdkClient').returns(Promise.resolve(clientSpy))
+        telemetryEnabledDefault = globals.telemetry.telemetryEnabled
```

Comment:
This looks like a pattern that could be encapsulated in a `testUtil.ts` function.

Reply:
yes, this is reusuable given how the current `codewhisperer.ts` calls the api. Will refactor this to testUtil, the next time we are adding unit tests around this.